src/mmds/src/data_store.rs

```diff
@@ -7,7 +7,7 @@ use serde_json::Value;
 
 /// The Mmds is the Microvm Metadata Service represented as an untyped json.
 #[derive(Clone)]
-pub struct Mmds {
+pub struct MmdsDataStore {
```

Comment:
Hmmm, naming this DataStore is a bit weird because you have a `DataStore` structure in the `data_store` module. The fact that `Mmds` is a data-store is a given because of the namespace (in this case the `data_store` module) from which you use it.

Reply:
I understand your logic. I've changed it back to how it was, and I do not have a strong suggestion right now. At the moment, we can let it be, and get back to it once we feel a better motivation to take action. Changes related to this aspect are not in the scope of this PR. A little bit of background for why I chose to rename it: 

In this specific `mmds::data_store` case, `Mmds` struct is the only struct inside the module. Maybe it was designed with this in mind. However, at a first look, `Mmds` name has a strong resonance and I can't associate it with a data store, because of what it means: `MicroVM metadata service`. To work around this naming issue, I think we can stop this namespace hierarchy at `mmds` level, defining a struct called `DataStore` or `MmdsDataStore` inside it, to have suggestive naming but also clear separation through namespaces. If we would need to define structs specific to the data store (e.g structs that compose a data store), then I think it would make sense to add another level in the namespace hierarchy, called `data_store`. It is only my thought process, and it is not backed by any data which proves that it is better to have it this way. The only data/anecdote is the clarity which will bring to me. When I'll need to use, and see it, I'll ask myself: `Ok, Mmds struct from mmds::data_store namespace, but what it has to do with the data store, and how can I use it?` Pretty common question, but imagine if we would have it named `mmds::MmdsDataStore`, or simply `mmds::DataStore`. I'll say then: `Oh, DataStore struct. This must be the MMDS data store. I do not know how to use it, yet, but I know what it is designed for.`